File: src/backend/internal/domain/project/phase5.go
package project

import (
	"time"

	"go.mongodb.org/mongo-driver/v2/bson"
)

type CodeFile struct {
	ID          bson.ObjectID `bson:"_id,omitempty" json:"id"`
	ProjectID   bson.ObjectID `bson:"project_id" json:"project_id"`
	Path        string        `bson:"path" json:"path"`
	Content     string        `bson:"content" json:"content"`
	TaskID      string        `bson:"task_id" json:"task_id"`
	Language    string        `bson:"language" json:"language"`
	Version     time.Time     `bson:"version" json:"version"`
	PhaseNumber int           `bson:"phase_number" json:"phase_number"`
	CreatedAt   time.Time     `bson:"created_at" json:"created_at"`
	UpdatedAt   time.Time     `bson:"updated_at" json:"updated_at"`
}

type CodeSymbol struct {
	Name    string `json:"name"`
	Kind    string `json:"kind"`
	Source  string `json:"source"`
	Backend bool   `json:"backend"`
}

type CodeContextManifest struct {
	Files            []CodeContextFile `json:"files"`
	Symbols          []CodeSymbol      `json:"symbols"`
	Dependencies     []string          `json:"dependencies"`
	EnvironmentHints []string          `json:"environment_hints"`
	ApproxTokens     int               `json:"approx_tokens"`
}

type CodeContextFile struct {
	Path     string `json:"path"`
	Language string `json:"language"`
	Purpose  string `json:"purpose"`
}

type Phase5Summary struct {
	TotalTasks            int64   `json:"total_tasks"`
	DoneTasks             int64   `json:"done_tasks"`
	InProgressTasks       int64   `json:"in_progress_tasks"`
	BlockedTasks          int64   `json:"blocked_tasks"`
	TodoTasks             int64   `json:"todo_tasks"`
	BackendFiles          int64   `json:"backend_files"`
	FrontendFiles         int64   `json:"frontend_files"`
	GeneratedLinesOfCode  int64   `json:"generated_lines_of_code"`
	AverageTaskMinutes    float64 `json:"average_task_minutes"`
	AutoRejections        int64   `json:"auto_rejections"`
	TotalPhaseTokens      int64   `json:"total_phase_tokens"`
	ExecutionMode         string  `json:"execution_mode"`
	CompletionPercent     float64 `json:"completion_percent"`
	LastExecutionUnixTime int64   `json:"last_execution_unix_time,omitempty"`
}

type CatastrophicFailureReport struct {
	CoveragePercent      float64  `json:"coverage_percent"`
	MaxCVSS              float64  `json:"max_cvss"`
	CredentialsExposed   bool     `json:"credentials_exposed"`
	CompilationFailed    bool     `json:"compilation_failed"`
	FailureDescriptions  []string `json:"failure_descriptions"`
	SourcePhase          int      `json:"source_phase"`
	RequestedBySystemTag string   `json:"requested_by_system_tag"`
}

File: src/frontend/src/types/phase5.ts
import { Task } from "./task";

export type Phase5ExecutionMode = "AUTOMATIC" | "MANUAL";

export interface Phase5Summary {
  total_tasks: number;
  done_tasks: number;
  in_progress_tasks: number;
  blocked_tasks: number;
  todo_tasks: number;
  backend_files: number;
  frontend_files: number;
  generated_lines_of_code: number;
  average_task_minutes: number;
  auto_rejections: number;
  total_phase_tokens: number;
  execution_mode: Phase5ExecutionMode;
  completion_percent: number;
  last_execution_unix_time?: number;
}

export interface Phase5CodeFile {
  id: string;
  project_id: string;
  path: string;
  content: string;
  task_id: string;
  language: string;
  version: string;
  phase_number: number;
  created_at: string;
  updated_at: string;
}

export interface Phase5CodeContextFile {
  path: string;
  language: string;
  purpose: string;
}

export interface Phase5CodeSymbol {
  name: string;
  kind: string;
  source: string;
  backend: boolean;
}

export interface Phase5CodeContext {
  files: Phase5CodeContextFile[];
  symbols: Phase5CodeSymbol[];
  dependencies: string[];
  environment_hints: string[];
  approx_tokens: number;
}

export interface Phase5FileListResponse {
  items: Phase5CodeFile[];
}

export interface Phase5TaskExecutionEvent {
  timestamp: string;
  taskId?: string;
  taskTitle?: string;
  level: "INFO" | "SUCCESS" | "ERROR";
  message: string;
}

export interface Phase5TaskWithFile extends Task {
  generatedFile?: Phase5CodeFile;
}
